src/main/java/org/apache/commons/collections4/bloomfilter/SetOperations.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Implementations of set operations on Bloom filters.
+ *
+ */
+public final class SetOperations {
+
+    /**
+     * Do not instantiate.
+     */
+    private SetOperations() {}
+
+    /**
+     * Verifies the Bloom filters have the same shape.
+     *
+     * @param first the first filter to check.
+     * @param second the second filter to check.
+     * @throws IllegalArgumentException if the shapes are not the same.
+     */
+    private static void verifyShape(BloomFilter first, BloomFilter second) {
+        if (!first.getShape().equals(second.getShape())) {
+            throw new IllegalArgumentException(String.format("Shape %s is not the same as %s",
+                first.getShape(), second.getShape()));
+        }
+    }
+
+    /**
+     * Calculates the Hamming distance between two Bloom filters.
+     *
+     * @param first the first Bloom filter.
+     * @param second the second Bloom filter.
+     * @return the Hamming distance.
+     */
+    public static int hammingDistance(BloomFilter first, BloomFilter second) {
+        verifyShape(first,second);
+        return first.xorCardinality(second);
+    }
+
+
+    /**
+     * Calculates the Jaccard similarity between two Bloom filters.
+     *
+     * <p>Also known as Jaccard index, Intersection over Union, and Jaccard similarity coefficient</p>
+     *
+     * @param first the first Bloom filter.
+     * @param second the second Bloom filter.
+     * @return the Jaccard similarity.
+     */
+    public static double jaccardSimilarity(BloomFilter first, BloomFilter second) {
+        verifyShape(first,second);
+        int orCard = first.orCardinality(second);
+        // if the orCard is zero then the hamming distance will also be zero.
+        return orCard==0?0:hammingDistance(first,second) / (double) orCard;
+    }
+
+    /**
+     * Calculates the Jaccard distance between two Bloom filters.
+     *
+     * <p>Jaccard distance is defined as {@code 1 - Jaccard similarity}</p>
+     *
+     * @param first the first Bloom filter.
+     * @param second the second Bloom filter.
+     * @return the Jaccard distance.
+     */
+    public static double jaccardDistance(BloomFilter first, BloomFilter second) {
+        return 1.0 - jaccardSimilarity(first,second);
+    }
+
+    /**
+     * Calculates the Cosine similarity between two Bloom filters.
+     * <p> Also known as Orchini similarity and the Tucker coefficient of congruence or
+     * Ochiai similarity.</p>
+     *
+     * <p>If either filter is empty (no enabled bits) the result is 0 (zero)</p>
+     *
+     * @param first the first Bloom filter.
+     * @param second the second Bloom filter.
+     * @return the Cosine similarity.
+     */
+    public static double cosineSimilarity(BloomFilter first, BloomFilter second) {
+        verifyShape(first,second);
+        int numerator = first.andCardinality(second);
+
+        return numerator==0?0:numerator / (Math.sqrt(first.cardinality()) * Math.sqrt(second.cardinality()));
```

Comment:
There is a cosine similarity in commons-text too I think, but probably not worth adding as another dependency (and I'm not even sure if that's dynamic enough to be used for comparing bloom filters)